SalesforceFoundation/k12-architecture-kit
# Apex code to sync Contact's Current Grade Level with their current Grade Enrollment

# Critical Changes

# Changes
We've added a TDTM class (GRER_SyncGradeLevel_TDTM.cls) that matches the Current Grade Level field on a Contact record to the Grade Level set on a Grade Enrollment record when its status is set to "Current." If any other Grade Enrollment record has a status of "Current" when another Grade Enrollment record is set to "Current," this class sets the status for that record to "Former."
Note the following behaviors for this new class:
- If you change the status of a Grade Enrollment record to "Current," the Grade Level Field on the Contact record is updated, as expected, but if you change the status of all Grade Enrollment records to "Former," the Contact record still displays a value for Current Grade Level, even though no Grade Enrollment record has a status of "Current."
- If you change the Current Grade Level field on a Contact record, the setting on the Grade Enrollment record is not updated.

Both of these issues will be addressed in an upcoming update.

Get more information on the GRER_SyncGradeLevel_TDTM.cls [here](https://github.com/SalesforceFoundation/Athena/blob/master/src/classes/tbd.cls).
# Issues Closed
Fixes #119 
# New Metadata

# Deleted Metadata

# Testing Notes
1. Create a test contact
2. Create a grade enrollment under contact grade enrollment related list. Fill in status as 'Current' and grade level as anything
3. Save the grade enrollment and make sure contact.current grade level is matching the grade level you just created
4. Create another grade enrollment under the same contact. Fill in status as 'Current' and grade level as something different than the first grade enrollment. 
5. Save the grade enrollment and make sure contact.current grade level is matching the grade level you just created and the first grade enrollment is now 'Former'
6. Update the grade level of the second grade enrollment to something else. 
7. Save and make sure the contact.current grade level has been updated accordingly
